src/lib/engines/layerCompositor/LayerCompositor.ts

```diff
@@ -302,18 +317,29 @@ export class LayerCompositor {
       `Canvas Camera: height ( ${canvasSize.height} ), center ( ${scaledImageCenter.x}, ${scaledImageCenter.y} ) `,
     );*/
 
-    const planeToImage = makeMatrix4Scale(new Vector3(canvasImageSize.width, canvasImageSize.height, 1.0));
+    const localToView = makeMatrix4Scale(new Vector3(canvasImageSize.width, canvasImageSize.height, 1.0));
 
     const offscreenScaledSize = this.offscreenSize.clone().multiplyByScalar(imageToCanvasScale);
-    const viewToLayerUv = makeMatrix3FromViewToLayerUv(offscreenScaledSize, undefined, true);
+    let viewToLayerUv = makeMatrix3FromViewToLayerUv(offscreenScaledSize, undefined, true);
+
+    // Apply crop translation (scale is handled via croppedImageSize)
+    viewToLayerUv = makeMatrix3Concatenation(
+      makeMatrix3Translation(
+        new Vector2(
+          this.cropOffset.x * +(this.imageSize.x / this.offscreenSize.x),
+          this.cropOffset.y * -(this.imageSize.y / this.offscreenSize.y),
+        ),
```

Comment:
Hate to admit it, but I guessed this math... I figured it out through trial and error. I did test thoroughly, but if there's one place to pay due diligence to in the code changes, it's this